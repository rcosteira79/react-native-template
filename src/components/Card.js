import React from 'react';
import { View } from 'react-native';

const Card = props => {
  return <View style={[styles.containerStyle, props.style]}>{props.children}</View>;
};

const styles = {
  containerStyle: {
    borderWidth: 1,
    borderRadius: 5,
    borderColor: 'white',
    borderBottomWidth: 0,
    shadowColor: 'black',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.24,
    shadowRadius: 2,
    elevation: 2,
    padding: 5,
    backgroundColor: 'white',
    justifyContent: 'flex-start',
    flexDirection: 'row'
  }
};

export { Card };
