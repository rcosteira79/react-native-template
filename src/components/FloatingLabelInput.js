import React from 'react';
import { View, TextInput, Animated } from 'react-native';

import EStyleSheet from 'react-native-extended-stylesheet';

class FloatingLabelInput extends React.Component {
  state = {
    isFocused: false
  };

  componentWillMount() {
    this.animatedIsFocused = new Animated.Value(this.props.value === '' ? 0 : 1);
  }

  componentDidUpdate() {
    Animated.timing(this.animatedIsFocused, {
      toValue: this.state.isFocused || this.props.value !== '' ? 1 : 0,
      duration: 200
    }).start();
  }

  handleFocus = () => this.setState({ isFocused: true });
  handleBlur = () => this.setState({ isFocused: false });

  handleAnimation(outputRange) {
    return this.animatedIsFocused.interpolate({
      inputRange: [0, 1],
      outputRange
    });
  }

  render() {
    const { label, unfocusedColor, focusedColor, refField, style, ...props } = this.props;
    const { viewStyle, labelStyle, textStyle } = styles;
    const labelStyleAnimations = {
      top: this.handleAnimation([35, 12]),
      fontSize: this.handleAnimation([14, 10]),
      color: this.handleAnimation([unfocusedColor, focusedColor])
    };

    const textStyleAnimations = {
      // Apparently, this can't be animated like the rest?
      borderBottomColor:
        this.state.isFocused || this.props.value !== '' ? focusedColor : unfocusedColor
    };

    return (
      <View style={[viewStyle, style]}>
        <Animated.Text style={[labelStyle, labelStyleAnimations]}>{label}</Animated.Text>
        <TextInput
          {...props}
          ref={refField}
          style={[textStyle, textStyleAnimations]}
          underlineColorAndroid={'transparent'}
          onFocus={this.handleFocus}
          onBlur={this.handleBlur}
        />
      </View>
    );
  }
}

const styles = EStyleSheet.create({
  viewStyle: {
    alignSelf: 'stretch',
    paddingTop: 18
  },
  labelStyle: {
    position: 'absolute',
    height: 40,
    fontSize: 14,
    fontFamily: 'Montserrat-Light',
    left: 0
  },
  textStyle: {
    height: 40,
    fontSize: 14,
    fontFamily: 'Montserrat-Light',
    color: '$colorGrey',
    borderBottomWidth: 1
  }
});

export { FloatingLabelInput };
