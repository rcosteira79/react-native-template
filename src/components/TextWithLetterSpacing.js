import React from 'react';
import { View } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';

import { Letter } from './Letter';

const spacingForLetterIndex = (letters, index, spacing) =>
  letters.length - 1 === index ? 0 : spacing;

export const TextWithLetterSpacing = props => {
  const { children, spacing, viewStyle, textStyle } = props;
  const letters = children.split('');

  return (
    <View style={[styles.container, viewStyle]}>
      {letters.map((letter, index) => (
        <Letter
          key={index}
          spacing={spacingForLetterIndex(letters, index, spacing)}
          textStyle={textStyle}
        >
          {letter}
        </Letter>
      ))}
    </View>
  );
};

const styles = EStyleSheet.create({
  container: {
    flexDirection: 'row'
  }
});
