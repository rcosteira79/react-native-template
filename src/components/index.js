export * from './FloatingLabelInput';
export * from './Card';
export * from './Button';
