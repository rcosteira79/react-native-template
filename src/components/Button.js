import React from 'react';
import EStyleSheet from 'react-native-extended-stylesheet';

import { TouchableOpacity } from 'react-native';
import { TextWithLetterSpacing } from './TextWithLetterSpacing';

const getButtonColor = enabled => {
  return EStyleSheet.value(
    enabled ? '$colorPrimary' : '$colorPrimaryDarkDisabled'
  );
};

const Button = ({ onPress, children, style, enabled }) => {
  const { buttonStyle, textViewStyle, textStyle } = styles;

  return (
    <TouchableOpacity
      style={[buttonStyle, style, { backgroundColor: getButtonColor(enabled) }]}
      disabled={!enabled}
      onPress={onPress}
    >
      <TextWithLetterSpacing
        spacing={2.4}
        viewStyle={textViewStyle}
        textStyle={textStyle}
      >
        {children}
      </TextWithLetterSpacing>
    </TouchableOpacity>
  );
};

const styles = EStyleSheet.create({
  textStyle: {
    color: 'white',
    fontSize: 14,
    fontFamily: 'Montserrat-Medium',
    letterSpacing: 2.4
  },
  textViewStyle: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  buttonStyle: {
    height: 50,
    borderRadius: 3,
    alignSelf: 'stretch'
  }
});

export { Button };
