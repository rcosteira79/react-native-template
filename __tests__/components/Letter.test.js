import React from 'react';
import renderer from 'react-test-renderer';

import { Letter } from '../../src/components/Letter';

jest.unmock();

describe('Letter component', () => {
  it('Is rendered correctly', () => {
    const tree = renderer
      .create(
        <Letter key={0} spacing={0} textSyle={{ flex: 1 }}>
          T
        </Letter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
