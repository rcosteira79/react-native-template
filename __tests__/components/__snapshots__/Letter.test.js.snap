// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Letter component Is rendered correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      Object {
        "paddingRight": 0,
      },
    ]
  }
>
  T
</Text>
`;
